### PR TITLE
doc(website): Add docs for plugin developers

### PR DESCRIPTION
**Pull request by hermanschaaf**
Some more detailed documentation about writing source plugins. There is more to say, but this is a start.

### CODE REVIEW

File: website/pages/docs/developers/creating-new-plugin.md [hunk 1]
```diff
@@ -1,18 +1,388 @@
...
+
+### Incremental Tables
+
+Some APIs lend themselves to being synced incrementally: rather than fetch all past data on every sync, an incremental table will only fetch data that has changed since the last sync. This is done by storing some metadata in a state **backend**. The metadata is known as a **cursor**, and it marks where the last sync ended, so that the next sync can resume from the same point. Incremental syncs can be vastly more efficient than full syncs, especially for tables with large amounts of data. This is because only the data that's changed since the last sync needs to be retrieved, and in many cases this is a small subset of the overall dataset.
```
Comment (disq) [inline website/pages/docs/developers/creating-new-plugin.md:43]: Could add link to the incremental docs here somewhere.

File: website/pages/docs/developers/creating-new-plugin.md [hunk 2]
```diff
@@ -1,18 +1,388 @@
...
+### Testing the Plugin
+
+There are two options for running a plugin before as a developer before it is released: as a gRPC server, or as a standalone binary.
+
+#### Run the Plugin as a gRPC Server
+
+This mode is especially useful for setting breakpoints your code for debugging, as you can run it in server mode from your IDE and attach a debugger to it. To run the plugin as a gRPC server, you can run the following command in the root of the plugin directory:
+
+```shell
+go run main.go serve
```
Comment (disq) [inline website/pages/docs/developers/creating-new-plugin.md:194]: I always do `go run . serve` which I find easier to write (or search in my bash history)

File: website/pages/docs/developers/creating-new-plugin.md [hunk 3]
```diff
@@ -1,18 +1,388 @@
...
+
+In the above example, we used a `Client` struct that we haven't talked about yet. The `Client` struct is used to store any data that needs to be shared between resolvers. For example, it may store the API key that we'll need to make API calls, or an SDK client that we'll use to make API calls. The `Client` struct is defined in the `client` directory, and is instantiated with a call to `client.New` in the `plugin` directory. In this case, we were using it to store an instance of the XKCD client. (We won't show the full XKCD client implementation here.)
+ 
+### Testing the Plugin
+
+There are two options for running a plugin before as a developer before it is released: as a gRPC server, or as a standalone binary.
```
Comment (disq) [inline website/pages/docs/developers/creating-new-plugin.md:187]: Here's a short doc on this https://www.cloudquery.io/docs/developers/running-locally

**Review by yevgenypats — APPROVED** [review]
🚀

File: website/pages/docs/developers/creating-new-plugin.md (hunk 3 above)
Comment (hermanschaaf) [reply website/pages/docs/developers/creating-new-plugin.md:187]: Ah thanks, I forgot about this.

File: website/pages/docs/developers/creating-new-plugin.md [hunk 4]
```diff
@@ -1,18 +1,388 @@
...
+The exact format doesn't matter, as long as it is unique for every multiplexed value.
+
+Now we can instruct the plugin SDK to use this multiplexer, where appropriate, by setting the `Multiplex` property on the table to `client.AccountMultiplex`:
+
+```go
+func MyTable() *schema.Table {
+    return &schema.Table{
+        Name:     "sample_table",
+        Resolver: fetchSampleTable,
+		Multiplex: client.AccountMultiplex,
```
[the remark this answers is not in the document]
Comment (hermanschaaf) [reply website/pages/docs/developers/creating-new-plugin.md:345]: Good catch!

File: website/pages/docs/developers/creating-new-plugin.md [hunk 5]
```diff
@@ -333,28 +333,28 @@ func (c *Client) ID() string {
 }
 ```
 
-The exact format doesn't matter, as long as it is unique for every multiplexed value.
+The exact format doesn't matter, as long as it is unique for every multiplexed value. Some plugins also put include `spec.Name` in the ID, to help identify the plugin in scenarios where multiple instances are run in parallel.
```
Comment (disq) [inline website/pages/docs/developers/creating-new-plugin.md:336]: `put include` 🤔

Comment (hermanschaaf) [reply website/pages/docs/developers/creating-new-plugin.md:336]: Oops, fixing

**Pull request merged**